evap/contributor/forms.py

```diff
@@ -73,6 +89,19 @@ def clean(self):
             self.add_error("vote_start_datetime", "")
             self.add_error("vote_end_date", _("The first day of evaluation must be before the last one."))
 
+        # Ensure locked questionnaires cannot be deselected and only unlocked ones can be added
+        selected_questionnaires = self.instance.general_contribution.questionnaires.filter(is_locked=True).distinct()
+        selected_questionnaires |= self.cleaned_data.get("general_questionnaires").filter(is_locked=False)
+        selected_questionnaires |= self.cleaned_data.get("dropout_questionnaires").filter(is_locked=False)
+
+        self.cleaned_data.update(
```

Comment:
I find it a bit odd that we update the form values instead of throwing a `ValidationError` when things don't add up. In person, we discussed that we can leave it as is for this PR because it is the same behavior as before, but I think we should revisit this afterwards. What do you think @richardebeling @Kakadus ?

Reply:
The UI already prevents selecting/deselecting locked questionnaires. But the final check can't be on client site.